src/serac/numerics/functional/tensor.hpp

```diff
@@ -1064,6 +1064,22 @@ SERAC_HOST_DEVICE constexpr auto dev(const tensor<T, n, n>& A)
   return devA;
 }
 
+/**
+ * @brief Returns a square matrix (rank-2 tensor) containing the diagonal entries of the input square matrix
+ * with zeros in the off-diagonal positions
+ * @param[in] A The input square matrix
+ * This operation is used to compute a term in the constitutive response of a linear, cubic solid material
+ */
+template <typename T, int n>
+SERAC_HOST_DEVICE constexpr auto diag(const tensor<T, n, n>& A)
+{
+  auto D = make_tensor<n, n>([](int, int) {return 0.0;});
```

Comment:
other possible implementations (that produce identical assembly for the 3x3 case): https://godbolt.org/z/c8nxGP897

personally, I prefer the implementation of `diag2` and `diag3` slightly

Reply:
I'm okay with any of these implementations. I wasn't sure if the `tensor` constructor initialized values to 0, but it seems from looking at `diag2` that it does.

Reply:
thanks for the clarification, definitely saved me some future debugging time!